typed/typed.go

```diff
@@ -119,6 +119,10 @@ func (tv TypedValue) Merge(pso *TypedValue) (*TypedValue, error) {
 // match), or an error will be returned. Validation errors will be returned if
 // the objects don't conform to the schema.
 func (tv TypedValue) Compare(rhs *TypedValue) (c *Comparison, err error) {
+	return tv.CompareFields(rhs, nil)
+}
+
+func (tv TypedValue) CompareFields(rhs *TypedValue, callback func(prefix *fieldpath.Path) bool) (c *Comparison, err error) {
```

Comment:
I don't know if I'd make this an arbitrary callback rather than give it a set of fields?

Reply:
Yeah I am inclined to agree that passing in a set of fields to skip is probably best?